client/src/components/Rooms/Rooms.js

```diff
@@ -20,55 +20,111 @@ class Rooms extends Component {
   state = {
     category: "",
     date: null,
-    showDate: true
+    showDate: true,
+    params: null
   };
+
+  nextPage = {
+    params: {
+      page: this.props.page
+    }
+  }
+
+  onScroll = () => {
+    if (this.refs.iScroll.scrollTop + this.refs.iScroll.clientHeight >= this.refs.iScroll.scrollHeight - 150
```

Comment:
Just curious, what is this.refs.iScroll? It's some npm package? react-scroll?

Reply:
Not npm package & not react-scroll...
One of use refs is 'catching' events when DOM was mounted. "iScroll" it's just custom name for refs:
...<div className="rooms-page" ref="iScroll">...
this.refs.iScroll.addEventListener("scroll", this.onScroll) ==> works like document.addEventListener("scroll", this.onScroll)